exponentation_by_squaring/exponentation_by_squaring.c

```diff
@@ -0,0 +1,27 @@
+#include<stdio.h>
+
+/* Algorithm to calculate x raised to power k very efficiently
+*  Can be used as a great alternative to normal pow function
+*  base - base of the expression
+*  power - power of expression
+*/
+
+long double exponentation_by_squaring(double base, int power) {
+    if (power < 0)		// Negative power case
+        return exponentation_by_squaring(1 / base, -power);
+    else if (power == 0)	// Base case1
+        return 1;
+    else if (power == 1)	// Base case2
+        return base;
+    else if (power % 2 == 0)
+        return exponentation_by_squaring(base * base, power / 2);
+    else if (power % 2 == 1)
+        return exponentation_by_squaring(base * base, (power - 1) / 2);
```

Comment:
Logic seems to be wrong here.
Assume base=2, power=3. 
This code will give 4. 
You should do `return base * exponentation_by_squaring(base * base, (power - 1) / 2);`.